crates/bevy_math/src/direction.rs

```diff
@@ -38,6 +38,30 @@ impl std::fmt::Display for InvalidDirectionError {
     }
 }
 
+/// Checks that a vector with the given squared length is normalized.
+///
+/// Warns for small error with a length threshold of approximately `1e-4`,
+/// and panics for large error with a length threshold of approximately `1e-2`.
+///
+/// The format used for the logged warning is `"Warning: {warning} The length is {length}`,
+/// and similarly for the error.
+#[cfg(debug_assertions)]
+fn assert_is_normalized(message: &str, length_squared: f32) {
+    let length_error_squared = (length_squared - 1.0).abs();
+
+    // Panic for large error and warn for slight error.
+    if length_error_squared > 2e-2 || length_error_squared.is_nan() {
+        // Length error is approximately 1e-2 or more.
+        panic!("Error: {message} The length is {}.", length_squared.sqrt());
```

Comment:
I think adding a \n newline before "The length" will make these error messages a bit more readable in practice.

Reply:
Not a very practical comparison, but just making it panic on purpose it looks like this without a newline and with a newline:

![kuva](https://github.com/bevyengine/bevy/assets/57632562/65e7c722-d1e7-4d3c-922f-e234c6daf3b8)

![kuva](https://github.com/bevyengine/bevy/assets/57632562/bd8b2f22-84d3-4851-9893-0c7948f96cb8)

Mostly preference, but having a newline there makes it not align with the error message so it almost looks like a separate message. In the case of warnings, it could also increase spam if you have a case where a direction is constantly slightly denormalized (because of physics, for example).